pkg/customresources/certificatestores/certificatestores.go

```diff
@@ -14,72 +14,82 @@ limitations under the License.
-func (c *ActiveCertStores) GetCertStores(_ string) map[string][]*x509.Certificate {
-	return c.ScopedCertStores[constants.EmptyNamespace]
+// It returns a list of certificates in the given store.
+func (c *ActiveCertStores) GetCertsFromStore(ctx context.Context, storeName string) []*x509.Certificate {
+	storeName = prependNamespaceToStoreName(storeName)
+	if !isCompatibleNamespace(ctx, storeName) {
+		return []*x509.Certificate{}
+	}
+	if certs, ok := c.scopedCertStores.Load(storeName); ok {
+		return certs.([]*x509.Certificate)
+	}
+	return []*x509.Certificate{}
 }
 
 // AddStore fulfills the CertStoreManager interface.
-// It adds the given certificate under the given scope.
-// TODO: Current implementation always adds the given certificate to cluster-wide cert store. Will support actual namespaced certStores in future.
-func (c *ActiveCertStores) AddStore(_, storeName string, certs []*x509.Certificate) {
-	scope := constants.EmptyNamespace
-	if c.ScopedCertStores[scope] == nil {
-		c.ScopedCertStores[scope] = make(map[string][]*x509.Certificate)
-	}
-	c.ScopedCertStores[scope][storeName] = certs
+// It adds the given certificate under cert store.
+func (c *ActiveCertStores) AddStore(storeName string, cert []*x509.Certificate) {
+	c.scopedCertStores.Store(storeName, cert)
 }
 
 // DeleteStore fulfills the CertStoreManager interface.
-// It deletes the certificate from the given scope.
-// TODO: Current implementation always deletes the cluster-wide cert store. Will support actual namespaced certStores in future.
-func (c *ActiveCertStores) DeleteStore(_, storeName string) {
-	if store, ok := c.ScopedCertStores[constants.EmptyNamespace]; ok {
-		delete(store, storeName)
+// It deletes the given cert store.
+func (c *ActiveCertStores) DeleteStore(storeName string) {
+	c.scopedCertStores.Delete(storeName)
+}
+
+// Namespaced verifiers could access certStores in the same namespace or "default" namespace.
+// Cluster-wide verifier could access all certStores.
+// Note: the cluster-wide behavior is different from KMP as we need to keep the behavior backward compatible.
+func isCompatibleNamespace(ctx context.Context, storeName string) bool {
+	namespace := ctxUtils.GetNamespace(ctx)
+	if namespace == constants.EmptyNamespace {
+		return true
 	}
+	return strings.HasPrefix(storeName, namespace+constants.NamespaceSeperator) || strings.HasPrefix(storeName, defaultNamespace+constants.NamespaceSeperator)
 }
 
-// IsEmpty fulfills the CertStoreManager interface.
-// It returns true if there are no certificates.
-func (c *ActiveCertStores) IsEmpty() bool {
-	count := 0
-	for _, certStores := range c.ScopedCertStores {
-		count += len(certStores)
+// prependNamespaceToStoreName prepends namespace to store name if not already present.
+// If the namespace where Ratify deployed is not set, `default` namespace will be prepended. However, this case should never happen.
```

Comment:
Just want to double check this is consistent behavior from before. I thought I remembered it would append the Ratify Namespace configured?

Reply:
that's good point. I was also considering if I should prepend default in this case. I found we always set the RATIFY_NAMESPACE to the env var, so the `default` option should never happen. But it could be different if users manually updated their helm chart and removed the RATIFY_NAMESPACE env var. To be strictly consistent with v1.1, I would remove prepending default namespace. Thanks for calling it out!